config/filter.d/apache-botsearch.conf

```diff
@@ -23,10 +23,7 @@ before = apache-common.conf
 
 [Definition]
 
-prefregex = ^%(_apache_error_client)s (?:AH\d+: )?<F-CONTENT>.+</F-CONTENT>$
-
-failregex = ^(?:File does not exist|script not found or unable to stat): <webroot><block>(, referer: \S+)?\s*$
-            ^script '<webroot><block>' not found or unable to stat(, referer: \S+)?\s*$
+failregex = ^"<ADDR>".+ HTTP/\d\.\d" 404
```

Comment:
1. this RE is vulnerable due to catch-all (`.+`), we don't accept filters with unjustified catch-all's anymore (see also https://github.com/fail2ban/fail2ban/pull/1961#pullrequestreview-75940560 for similar PR);
2. this filter does not work so: it searching for certain bots only (declared in include botsearch-common.conf as `block`) for certain `webroot`; both parameters can be also set by user (supplied to filter in jail).
Your variant simply blocks 404 response (regardless the URI, bot-list and root), so I guess it is targeting totally different kind of purposes. I'm not against to make it belonging to this filter (but then not by default, but rather within some mode `aggressive`). 
Anyway to solve #2378 one must change log-format, not the functionality of the filter.

Reply:
Sorry for my late reply, had been knocked down by some customer projects. Still would like to get this sorted and I'd rather love to get the original filter being adjusted properly for the new Apache log format which was described in #2378

Now, with your comment above I'm not even sure anymore, if we're trying to tackle the correct problem here.

Trying to find out, let me show you what we have in the logs.

**In the error log we have lots of these:**

```
[Wed Mar 17 05:27:05.098401 2021] [error] [pid 9550] AH01071: Got error 'Primary script unknown'
[Wed Mar 17 05:27:05.303839 2021] [error] [pid 9550] AH01071: Got error 'Primary script unknown'
[Wed Mar 17 05:27:05.509085 2021] [error] [pid 9550] AH01071: Got error 'Primary script unknown'
[Wed Mar 17 05:27:05.714101 2021] [error] [pid 9550] AH01071: Got error 'Primary script unknown'
[Wed Mar 17 05:27:05.921099 2021] [error] [pid 9550] AH01071: Got error 'Primary script unknown'
```

Those are triggered by PHP FPM for all requests to PHP files that don't exist. Therefore, each of the above lines has a matching entry in the access log looking like this:

```
"1.9.204.148" - - [17/Mar/2021:05:27:05 +0100] "POST /q.php HTTP/1.1" 404 229 "-" "Mozilla/4.0 (compatible; MSIE 9.0; Windows NT 6.1; Trident/5.0)"
"1.9.204.148" - - [17/Mar/2021:05:27:05 +0100] "POST /pe.php HTTP/1.1" 404 229 "-" "Mozilla/4.0 (compatible; MSIE 9.0; Windows NT 6.1; Trident/5.0)"
"1.9.204.148" - - [17/Mar/2021:05:27:05 +0100] "POST /hm.php HTTP/1.1" 404 229 "-" "Mozilla/4.0 (compatible; MSIE 9.0; Windows NT 6.1; Trident/5.0)"
"1.9.204.148" - - [17/Mar/2021:05:27:05 +0100] "POST /sz.php HTTP/1.1" 404 229 "-" "Mozilla/4.0 (compatible; MSIE 9.0; Windows NT 6.1; Trident/5.0)"
"1.9.204.148" - - [17/Mar/2021:05:27:05 +0100] "POST /cainiao.php HTTP/1.1" 404 229 "-" "Mozilla/4.0 (compatible; MSIE 9.0; Windows NT 6.1; Trident/5.0)"
```

My take now is that those requests have nothing to do what is prepared in botsearch-common.conf but they are obviously bot search attempts.

Is apache-botsearch.conf the right approach to block those? If yes, can you give me some hints on how to geth there? If no, what other approach would be correct - if any?

Reply:
Had another look and it seems that apache-badbots is better suited for this task. I tested it with `fail2ban-regex -o row test-apache-access.log apache-badbots` but it doesn't pick up the line above and I can't figure out why.